cosmos/converter.py

```diff
@@ -181,6 +181,11 @@ def validate_adapted_user_config(
             "RenderConfig.dbt_project_path is required for rendering an airflow DAG from a DBT Graph if no manifest is provided."
         )
 
+    if render_config and render_config.project_path:
+        # Project config is further expected to have project name
+        # even if it had no manifest
+        project_config.project_name = render_config.project_path.stem
```

Comment:
Should we also check here that `project_config.project_name` is NOT already set and only then assign it from `render_config`?

Reply:
They are mutually exclusive... Cosmos errors if both of them are set. I can add a comment to this!

Reply:
Created a separate PR with this change, since it solves another underlining issue:
https://github.com/astronomer/astronomer-cosmos/pull/1307

Adding a proper test case for it